sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -368,7 +368,11 @@ export type ServiceBusErrorCode =
 /**
  * The user doesn't have access to the entity.
  */
- | "UnauthorizedAccess";
+ | "UnauthorizedAccess"
+/**
+ * The number of unsettled messages that can be held is reached, please settle the received messages.
+ */
+ | "UnsettledMessagesLimitExceeded";
```

Comment:
I believe our friends in .NET encounter the same situation. Can you check what kind of error they throw when a user has too many unsettled messages and so we cannot given them any more messages?

Reply:
We changed a link setting to avoid this - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs#L705

Reply:
Woah..

@ramya-rao-a I believe the equivalent for us would be to set the capacity of circular buffer instead of hard-coded 2048 at rhea.

Reply:
@JoshLove-msft In your link, you set the value to Max, were there no concerns w.r.t memory consumption?
Also, this means that the users are allowed to never settle the messages. Isn't that a bad practice?

Reply:
Let's also ask our Java and Python friends :)

Reply:
In terms of memory consumption, if there is some app that only wants to settle messages under very specific circumstances like 1 out of 100 messages received, then yes I would expect that they would see a large memory footprint. There is apparently still a service-side limit that is enforced, so I don't think it would get anywhere close to the uint.MaxValue in practice - https://github.com/Azure/azure-sdk-for-net/pull/16163#issue-508025290

Yes, it is recommended to settle messages, but this was the approach that the service team settled on in Track 1 to unblock customers and we copied it for Track 2.

Reply:
In java we are not putting any limit on unsettled messages in SDK.  
By default SDK settle the message. When user shows intention to handle settling manually by them selves by disabling auto complete, we should trust them.

Reply:
Thanks for the link @JoshLove-msft, that's good info.

Based on the link the number of messages at max can be 5000. 
Just curious.. @JoshLove-msft @hemanttanwar.. have you guys seen any weird behavior at the terminal state.. such as 5000 outstanding unsettled messages.

**Cases worth checking** - receive 5000 messages in peeklock mode and not settle them (for both Streaming receiver and Batching receiver)

_PS: This PR was meant to fix a behavior at such terminal state that we have seen with the JS SDK. 
More info for reference - https://github.com/Azure/azure-sdk-for-js/issues/11633#issuecomment-712305487._

Reply:
I think the link is saying that prior to this change 5000 was the max. We have a test where we have 6000 unsettled messages - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs#L878

Reply:
Thanks @JoshLove-msft, makes sense!

Reply:
@yunhaoling mentioned the window in python is 64*1024, and he is observing some weirdness around receiving 50K messages, he's testing for more observations.

Reply:
@HarshaNalluru Based on what other languages are doing, are we planning to update rhea to support removing the 2048 message limit?

Reply:
@chradek yes, we'd want want it to be configurable, I'll make an issue soon.
(The current PR is still required since the issues present at the edge cases are independent of the value 2048.)